### PR TITLE
[WIP] Add api-gateway support.

### PR DESCRIPTION
**Pull request by lzbj**
Address issue #187.

### CODE REVIEW

File: deployers/servicedeployer.go [hunk 1]
```diff
@@ -397,8 +407,55 @@ func (deployer *ServiceDeployer) createAction(pkgname string, action *whisk.Acti
...
+		// create the API gateway according to the action definition in the manifest.yaml
+		// if no errors happened during the action creation.
+		// since the spec for this is not publish, so comment below lines.
+
+		//_, _, err := deployer.Client.Apis.Insert(createApiEntity(deployer, action), true)
+		//if err != nil {
+		//	wskErr := err.(*whisk.WskError)
+		//	fmt.Printf("Got error creating API with error message: %v and error code: %v.\n", wskErr.Error(), wskErr.ExitCode)
+		//}
+
```
Comment (paulcastro) [inline deployers/servicedeployer.go]: Is this empty block a placeholder for something?

Comment (lzbj) [inline deployers/servicedeployer.go]: Perhaps not a placeholder, since if we need support the api gateway full grammer, I'll remove it.

**Comment by paulcastro** [comment]
@lzbj rebase with master?

**Comment by lzbj** [comment]
@paulcastro , sure, I'll perhaps update with a full grammar support, and then rebase master.

**Pull request closed without merging**